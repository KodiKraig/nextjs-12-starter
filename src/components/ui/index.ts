export { default as Logo } from './Logo';
export { default as Button } from './Button';
export { default as Text } from './Text';
export { default as Input } from './Input';
export { default as SectionHeader } from './SectionHeader';
