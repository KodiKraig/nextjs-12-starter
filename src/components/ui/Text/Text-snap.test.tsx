import * as React from 'react';
import { Text } from '@components/ui';
import renderer from 'react-test-renderer';

describe('Text renders different variants', () => {
  it('Body', () => {
    const component = renderer.create(
      <Text style={{ alignItems: 'center' }}> Hello World </Text>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Heading', () => {
    const component = renderer.create(
      <Text variant="heading"> Hello World </Text>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Section Heading', () => {
    const component = renderer.create(
      <Text variant="sectionHeading"> Hello World </Text>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Page Heading', () => {
    const component = renderer.create(
      <Text variant="pageHeading"> Hello World </Text>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
