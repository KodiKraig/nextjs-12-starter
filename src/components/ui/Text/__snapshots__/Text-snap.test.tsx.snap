// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders different variants Body 1`] = `
<div
  className="root body"
  style={
    {
      "alignItems": "center",
    }
  }
>
   Hello World 
</div>
`;

exports[`Text renders different variants Heading 1`] = `
<h1
  className="root heading"
>
   Hello World 
</h1>
`;

exports[`Text renders different variants Page Heading 1`] = `
<h2
  className="root pageHeading"
>
   Hello World 
</h2>
`;

exports[`Text renders different variants Section Heading 1`] = `
<h3
  className="root sectionHeading"
>
   Hello World 
</h3>
`;
